signals.c

```diff
@@ -0,0 +1,72 @@
+/*
+  Not part of Grbl. KeyMe specific.
+  
+  Signal is one layer of abstraction above adc.h and adc.c.
+
+  ADC values are read in specific time intervals, filtered and
+  stored in the appropriate arrays.
+*/
+
+#include "signals.h"
+#include "adc.h"
+#include "systick.h"
+
+#define N_FILTER 3
+#define SIGNALS_CALLBACK_INTERVAL 20
```

Comment:
The sampling rate of the ADC depends on a number of factors. We are also limited as to how fast we can sample.

* The ADC sampling rate depends on the speed of the gripper (i.e. the frequency of the stepper ISR).
* It also depends on the accuracy of the ADC. We only have a 10-bit ADC. If the value of the ADC changes, for example, every 200 steps of the stepper ISR, it is unncessary to sample in between gripper motor steps.
* I agree that in an ideal situation, we would like to get filtered, up-to-date information in between steps to prevent the gripper from overshooting by a few steps. In order to achieve this, we need to sample the ADC at the same frequency or faster than the stepper ISR. However, since our micro-controller does not support interrupt priorities, if we do so we run the risk of having the critical stepper ISR frequently waiting for the non-critical soft-real-time sys_tick ISR which controls the ADC sampling rate.

The goal of this pull request is to implement ADC filtering for a future force feedback loop implementation, but not to have the sampling rate for the force feedback loop sorted out. 

 On another branch I have a force feedback loop which works without filtered ADC data. I suggest that we sort out the ADC sampling rate for the force feedback loop once we implement the force feedback loop after this pull requests gets merged.

I would like to run tests to figure out how fast we should sample for the force feedback loop, but I am waiting for this branch to get reviewed and merged so that I can rebase my force loop branch off of the new 4axis which includes ADC filtering.